### PR TITLE
Allow for response errors to be passed back from a plugin

### PR DESCRIPTION
**Pull request by briankassouf**
Plugins were previously swallowing the response errors when an error was also returned:
```
return logical.ErrorResponse("..."), logical.ErrInvalidRequest
```

### CODE REVIEW

**Comment by calvn** [comment]
Should the same be applied for `HandleExistenceCheck()`?

**Comment by jefferai** [comment]
^ that -- other than that looks good. Although keep in mind that this may need to be applied to the persona lookahead once that is on this side of things, so you may want to do that as a followup on the other side so it doesn't get lost.

**Comment by briankassouf** [comment]
@jefferai @calvn I believe `HandleExistenceCheck` is already handling this correctly. Jeff, I'll make a PR for the persona lookahead version.

File: logical/plugin/backend_client.go [hunk 1]
```diff
@@ -117,10 +117,9 @@ func (b *backendPluginClient) HandleRequest(req *logical.Request) (*logical.Resp
 		if reply.Error.Error() == logical.ErrUnsupportedOperation.Error() {
 			return nil, logical.ErrUnsupportedOperation
 		}
-		return nil, reply.Error
```
Comment (calvn) [inline logical/plugin/backend_client.go]: I think you might need to do `return reply.Response, reply.Error` here, and keep the other line to explicitly return nil

**Comment by briankassouf** [comment]
@calvn updated, and I tested with the the kubernetes plugin

**Pull request merged**